content/posts/how-to/determine-operating-system.md

```diff
@@ -37,9 +37,85 @@ In your terminal, run `uname`.
 
 - If you see [TODO] then you are on a Windows machine. Go to [Step 5](#step-5)
 
+*NOTE* If you run this on a container, the output will refer to the host machine's operating system, not the container's (technically, the container doesn't have an operating system, it has a distribution, eg Debain).
+
 ### 3) Get Linux OS Details {#step-3}
 
-TODO
+### 3.1) Run `uname -a` {#step-3-1}
+
+Running this command will get you information about the running kernel:
+
+```shell
+uname -a
+```
+
+The example output:
+
+```
+Linux dali 5.4.0-37-generic #41-Ubuntu SMP Wed Jun 3 18:57:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
+```
+
+tells you (in order) the:
+
+- OS / Kernel name (`Linux` above)
+
+- Nodename / short hostname (`dali` above)
+
+- Kernel release (`5.4.0-37-generic` above)
+
+- Kernel version (`#41-Ubuntu SMP Wed Jun 3 18:57:02 UTC 2020` above)
+
+- Machine hardware name (`x86_64` above)
+
+- Processor type (`x86_64` above)
+
+- Hardware platform (`x86_64` above)
+
+- Operating system (`GNU/Linux` above)
+
+*NOTE* If you run this on a container, the output will refer to the host machine's operating system, not the container's (technically, the container doesn't have an operating system, it has a distribution, eg Debain).
+
+### 3.2) Get Linux OS Details {#step-3-2}
+
+If you have the `lsb_release` command available on your machine, you can run:
+
+```shell
+lsb_release -a
+```
+
+to get the following details:
+
+```
+Distributor ID: Ubuntu
+Description:    Ubuntu 20.04 LTS
+Release:        20.04
+Codename:       focal
+```
+
+If you don't have this command, then you can try installing it in your package manager (search for `lsb-core`, eg `centos-lsb-core`, or `lsb-release`). Confusingly, the command has an underscore, but the package name usually has dashes.
+
+### 3.3) Get Linux OS Distribution {#step-3-3}
+
+There are some `/etc` files that may give you more information:
+
+```shell
+cat /etc/issue
+cat /etc/issue.net
+cat /etc/os-release
+```
+
+### 3.4) Per-Distribution Information {#step-3-4}
+
+Depending on your distribution, more information may be gleaned.
+
+#### 3.4.1) RedHat / CentOS {#step-3-4-1}
+
+```shell
+cat /etc/centos-release
+cat /etc/redhat-release
+cat /etc/system-release
+cat /etc/system-release-cpe
+```
```

Comment:
:heart:

Reply:
There's plenty more distributions to cover, if you like that :)
I just ran up a docker container and searched through /etc...

Reply:
Oh interesting strategy. I was commenting on the entire doc though. I'll definitely end up there in the future. :D